File: evaluating_platform/frontend/src/services/assessment.ts
import api from './api'
import { authService } from './auth'

export interface Assessment {
  id: string
  user_id: string
  name: string
  description: string
  goal: string
  target_id: string
  template_id?: string
  status: 'pending' | 'running' | 'completed' | 'failed' | 'canceled'
  plan: string[]
  error_msg: string
  created_at: string
  updated_at: string
  completed_at?: string
}

export interface LogEvent {
  assessment_id: string
  iteration: number
  tool_name: string
  output: string
  severity?: string
  tokens_used: number
  duration_ms: number
  timestamp: string
  type?: string
}

export interface CreateAssessmentData {
  name: string
  goal: string
  target_type: string
  target_url: string
  target_key?: string
  target_model?: string
  template_id?: string
  description?: string
}

export const assessmentService = {
  async create(data: CreateAssessmentData): Promise<{ assessment_id: string; status: string }> {
    const res = await api.post('/assessments', data)
    return res.data
  },

  async list(limit = 20, offset = 0): Promise<{ items: Assessment[]; total: number }> {
    const res = await api.get('/assessments', { params: { limit, offset } })
    return res.data
  },

  async get(id: string): Promise<{ assessment: Assessment; report?: unknown }> {
    const res = await api.get(`/assessments/${id}`)
    return res.data
  },

  async cancel(id: string): Promise<void> {
    await api.post(`/assessments/${id}/cancel`)
  },

  streamLogs(
    assessmentID: string,
    onLog: (event: LogEvent) => void,
    onDone: () => void,
  ): () => void {
    const token = authService.getToken()
    const base = (import.meta as unknown as { env: { VITE_API_BASE?: string } }).env.VITE_API_BASE || 'http://localhost:8080/api/v1'
    const url = `${base}/assessments/${assessmentID}/stream?token=${encodeURIComponent(token || '')}`
    const es = new EventSource(url)

    es.onmessage = (e) => {
      try {
        const data: LogEvent = JSON.parse(e.data)
        if (data.type === 'done') {
          onDone()
          es.close()
        } else {
          onLog(data)
        }
      } catch {
        // ignore malformed
      }
    }
    es.onerror = () => {
      onDone()
      es.close()
    }

    return () => es.close()
  },
}

File: evaluating_platform/internal/model/assessment.go
package model

import (
	"time"

	"github.com/google/uuid"
)

// AssessmentStatus 评估任务状态
type AssessmentStatus string

const (
	StatusPending   AssessmentStatus = "pending"
	StatusRunning   AssessmentStatus = "running"
	StatusCompleted AssessmentStatus = "completed"
	StatusFailed    AssessmentStatus = "failed"
	StatusCanceled  AssessmentStatus = "canceled"
)

// Assessment 评估任务
type Assessment struct {
	ID          uuid.UUID        `json:"id" db:"id"`
	UserID      uuid.UUID        `json:"user_id" db:"user_id"`
	Name        string           `json:"name" db:"name"`
	Description string           `json:"description" db:"description"`
	Goal        string           `json:"goal" db:"goal"` // 自然语言评估目标
	TargetID    uuid.UUID        `json:"target_id" db:"target_id"`
	TemplateID  *uuid.UUID       `json:"template_id,omitempty" db:"template_id"`
	Status      AssessmentStatus `json:"status" db:"status"`
	Plan        []string         `json:"plan" db:"plan"` // LLM 规划的工具序列
	ErrorMsg    string           `json:"error_msg,omitempty" db:"error_msg"`
	CreatedAt   time.Time        `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time        `json:"updated_at" db:"updated_at"`
	CompletedAt *time.Time       `json:"completed_at,omitempty" db:"completed_at"`
}

// AssessmentLog 评估执行日志（每次工具调用记录）
type AssessmentLog struct {
	ID           uuid.UUID `json:"id" db:"id"`
	AssessmentID uuid.UUID `json:"assessment_id" db:"assessment_id"`
	Iteration    int       `json:"iteration" db:"iteration"`
	ToolName     string    `json:"tool_name" db:"tool_name"`
	Input        string    `json:"input" db:"input"`   // JSON
	Output       string    `json:"output" db:"output"` // JSON
	Thinking     string    `json:"thinking" db:"thinking"` // LLM 推理链
	TokensUsed   int       `json:"tokens_used" db:"tokens_used"`
	DurationMs   int64     `json:"duration_ms" db:"duration_ms"`
	CreatedAt    time.Time `json:"created_at" db:"created_at"`
}

// TargetSystem 被评估的目标系统
type TargetSystem struct {
	ID          uuid.UUID `json:"id" db:"id"`
	UserID      uuid.UUID `json:"user_id" db:"user_id"`
	Name        string    `json:"name" db:"name"`
	Type        string    `json:"type" db:"type"` // openai | agent | custom
	Endpoint    string    `json:"endpoint" db:"endpoint"`
	APIKey      string    `json:"api_key,omitempty" db:"api_key"`
	Config      string    `json:"config" db:"config"` // JSON，额外配置
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
}
